### ISSUE TITLE
Add the Eval SOP data stream to React DATIM User Management app

### ISSUE THREAD

**Issue by DavidNesbitt** [pepfar-datim/datim-user-management-ng#126]
Original ticket: https://github.com/pepfar-datim/Global/issues/4018

![image](https://user-images.githubusercontent.com/23367121/56004366-d764c280-5c98-11e9-984a-946962d6e6aa.png)

ALL DATIM users types, _except MOH users_, should have `Read` access to ESoP data (in event reports app; Adherence Checklist and Evaluation Checklist).

Example:
![image](https://user-images.githubusercontent.com/23367121/56004405-23b00280-5c99-11e9-8b51-8adbc7403edc.png)

As a note:
If a user receives read-only access, they should get `Data Entry Tracker` role and be added to the `Data ESOP access` group.

If a user receives read-write access, they should get `Data Entry Tracker` role and be added to both the `Data ESOP access` group and the `Data ESOP entry` group.  Read-write (Enter Data) for Eval SOP is conferred to only a few users, and approved by Kristin Kelling at OGAC.

@hkim224 to review and confirm
fyi: @jakub-bao

**Comment by benguaraldi** [comment]
@DavidNesbitt @hkim224 Given that all users have access to view Eval SOP (see [this comment](https://github.com/pepfar-datim/Global/issues/4016#issuecomment-431484735)), it seems to me that we don't need to assign any users to just `Eval SOP access`.  Do you agree?

Should the app also have a radio button for `Enter Data` for `Eval SOP`?  Or do we want that to be granted only in the DHIS2 Users app?  If we're skipping the `Enter Data`, I recommend not listing the stream, but instead just providing a message.

**Comment by DavidNesbitt** [comment]
Thanks @benguaraldi!  All users _except MOH users_ have access to `view ESoP`.

A message or notification, in lieu of the radio button mock-up in my first comment, is probably fine. I just want User Admins to be able to see/note/understand that ESoP is inherent to the Partner, Agency, IA, Agency HQ, and Global users (when they look-up or invite users they administer). Putting it under the Data Stream section just made "visual" sense to me, but your suggestion would meet the spirit of the requirement...so whichever is easier for you and @jakub-bao from a technical design POV should work.

We do not want a radio button for `Enter ESoP Data` since that is granted through an OGAC approval process.  If a user admin encounters a user that has the `Enter ESoP Data` permission, then the warning notification message should trigger (about not being able to edit that user because of having that special access - which I believe has already been added to the REACT version functionality)

**Comment by benguaraldi** [comment]
@DavidNesbitt, I made a related comment at:
https://github.com/pepfar-datim/Global/issues/4156#issuecomment-486261817

Please respond there.  Thanks!

**Comment by jakub-bao** [comment]
Won't fix per discussion with @hkim224

**Comment by benguaraldi** [comment]
@hkim224 I know this isn't MVP, but are you sure we don't want this in the future?  I suspect we do.

FYI, @hardingt

**Comment by hardingt** [comment]
@hkim224 agreed with Ben, are sure we are never going to implement this?

**Comment by benguaraldi** [comment]
Reopening so we don't lose the issue as we discuss.

**Comment by jakub-bao** [comment]
If I understand correctly:

1. The REACT app will add each new user to the group "Data ESOP access"
2. The REACT app will also grant each new user "Data Entry Tracker" role
3. We will need to run a script to add all existing users to the group and give them the role as well.

@hkim224 @benguaraldi @hardingt Can you confirm?

**Comment by benguaraldi** [comment]
I suspect that's incorrect, @jakub-bao, but I'm pretty confused at this point.  I don't think we've clarified whether MOH users get access to the ESOP data, which is a necessary first step to determining how we're going to implement.

Once we decide that, we probably want to split Data Entry Tracker into two roles, as I imagine some of the authorities are only needed when a user can enter data.  Alternatively, we can add those authorities to Read Only if it's okay for MOH users to see the ESOP data.

@hkim224, what's the final analysis with MOH users being able to see the ESOP data?

**Comment by hkim224** [comment]
- MOH users do NOT get access to ESOP data. noted by David above also: https://github.com/pepfar-datim/datim-user-management-ng/issues/126#issuecomment-486229621
- All other types should get read-only access to ESOP data. 
- The ability to EDIT ESOP data is given on an ad hoc basis and is approved by program.

**Comment by benguaraldi** [comment]
@hkim224 Roger.  I was awaiting confirmation from you on that in response to [this comment](https://github.com/pepfar-datim/Global/issues/4156#issuecomment-486267069).

I've opened up https://github.com/pepfar-datim/Global/issues/4990 to set up the roles and groups correctly.  Let's wait on this until the requirements on that ticket are clear.

**Comment by benguaraldi** [comment]
I believe this is complete.  Closing.
